dockerfiles/grist/ressources/dinum-custom.js

```diff
@@ -0,0 +1,143 @@
+    selectedTz: 'CEST',
+
+    // Date du début de la maintenance
+    startDateWithoutTimezone: '2024-04-18T17:30:00',
+
+    // Durée de la maitenance prévue (en minutes).
+    // ASTUCE: Si la maintenance dure moins longtemps que prévu, réduire a postériori la valeur de cette variable de sorte à ce qu'elle n'apparaisse plus.
+    // Pas besoin de committer ensuite le changement dans le repo.
+    maintainanceDurationMinutes: 60,
+  };
+  // FIN CONFIGURER
+
+  const tz = { CEST: '0200', CET: '0100' };
+
+  const maintainanceStartDate = new Date(`${config.startDateWithoutTimezone}+${tz[config.selectedTz]}`);
+  const maintainanceEndDate = (function () {
+    const date = new Date(maintainanceStartDate); // If I may, Date API is really ackward, especially because it is not immutable
+    date.setMinutes(maintainanceStartDate.getMinutes() + config.maintainanceDurationMinutes);
+    return date;
+  })();
+
+
+  function showDialog() {
+    const dialog = document.createElement('dialog');
+    dialog.id = 'maintenancePopin';
+    const dateFormatter = new Intl.DateTimeFormat('fr-FR', {
+      dateStyle: 'full',
+      timeStyle: 'short',
+      timeZone: 'Europe/Paris',
+    });
+    const timeFormatter = new Intl.DateTimeFormat('fr-FR', {
+      timeStyle: 'short',
+      timeZone: 'Europe/Paris',
+    });
+
+    dialog.innerHTML = `
+        <p>
+        Une maintenance de Grist est prévue le ${dateFormatter.format(maintainanceStartDate)} heure de Paris
+          jusqu'à ${timeFormatter.format(maintainanceEndDate)}, période durant laquelle le service sera momentanément indisponible. <br>
+        Pour toute remarque ou question, merci de nous contacter via <a href="mailto:donnees@anct.gouv.fr">donnees@anct.gouv.fr</a></p>
+        <p>Merci de votre compréhension</p>
+        <form method="dialog">
+          <p>
+            <input type="checkbox" id="jaiCompris">
+            <label for="jaiCompris">J'ai lu et compris, merci de ne plus me montrer ce message</label>
+          </p>
+          <button disabled id="fermerPopinMaintenance">Fermer</button>
+        </form>
+    `;
+    document.body.appendChild(dialog);
+    document.getElementById('fermerPopinMaintenance').onclick = function onMessageUnderstood() {
+      localStorage.maintainanceStartDateAgreement = maintainanceStartDate.toISOString();
+    }
+    document.getElementById('jaiCompris').onchange = function (ev) {
+      document.getElementById('fermerPopinMaintenance').disabled = !ev.target.checked;
+    }
+    dialog.showModal();
+  }
+
+  if (Date.now() < maintainanceEndDate.getTime() &&
+    (!localStorage.maintainanceStartDateAgreement || localStorage.maintainanceStartDateAgreement !== maintainanceStartDate.toISOString())) {
+    window.addEventListener('load', showDialog);
+  }
+})();
+
+// ========================
+// END dialog
+// ========================
+
+// ========================
+// START gauffre
+// ========================
+
+window.onload = (event) => {
```

Comment:
Prefer using `window.addEventListener`

```suggestion
window.addEventListener('load', (event) => {
```

This way, you will not have overriding issues to listen to this event.